packages/sources/alongside/src/endpoint/collateral/index.ts

```diff
@@ -103,32 +109,118 @@ export class AlongsideCollateralTransport implements Transport<EndpointTypes> {
     }
   }
 
-  async foregroundExecute(
-    req: AdapterRequest<EndpointTypes['Request']>,
-    settings: typeof config.settings,
-  ): Promise<AdapterResponse<EndpointTypes['Response']>> {
-    const requestTradingBalance = this.prepareRequest('TRADING', settings)
-    const requestTradingVault = this.prepareRequest('VAULT', settings)
-    const collateral = new Collateral(settings.RPC_URL)
+  async backgroundHandler(context: EndpointContext<EndpointTypes>): Promise<void> {
+    let tradingBalances
+    let vaultBalances
+    const collateral = new Collateral(context.adapterSettings.RPC_URL)
     const providerDataRequestedUnixMs = Date.now()
-    logger.debug('Requesting trading balance')
-    const tradingBalances = await this.requester.request<ProviderResponseBody>(
-      req.id,
-      requestTradingBalance,
-    )
-    logger.debug('Requesting trading vault')
-    const vaultBalances = await this.requester.request<ProviderResponseBody>(
-      req.id,
-      requestTradingVault,
-    )
-    logger.debug('Getting asset weights')
-    const units = await collateral.getAssetWeights()
-    logger.debug('Calculating minimum collateral')
-    const result = collateral.calcMinCollateral(
-      tradingBalances.response.data.balances,
-      vaultBalances.response.data.balances,
-      units,
-    )
+
+    try {
+      logger.debug('Preparing request for trading balance')
+      const requestTradingBalance = this.prepareRequest('TRADING', context.adapterSettings)
+      logger.debug('Requesting trading balance')
+      tradingBalances = await this.requester.request<ProviderResponseBody>(
+        'trading_balance_request',
+        requestTradingBalance,
+      )
+    } catch (e) {
+      const errorMessage = 'Error occurred retrieving trading balance'
+      logger.error(errorMessage)
+      await this.responseCache.write(this.name, [
+        {
+          params: {},
+          response: {
+            statusCode: 502,
+            errorMessage,
+            timestamps: {
+              providerDataRequestedUnixMs,
+              providerDataReceivedUnixMs: Date.now(),
+              providerIndicatedTimeUnixMs: undefined,
+            },
+          },
```

Comment:
Some of these requests can be parallelized, I know it's not where the big delay happens but let's do it anyways